smf/installinator/manifest.xml

```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0"?>
+<!DOCTYPE service_bundle SYSTEM "/usr/share/lib/xml/dtd/service_bundle.dtd.1">
+
+<service_bundle type='manifest' name='installinator'>
+
+<service name='system/illumos/installinator' type='service' version='1'>
+  <create_default_instance enabled='false' />
+  <single_instance />
+
+  <dependency name='multi_user' grouping='require_all' restart_on='none'
+    type='service'>
+  <service_fmri value='svc:/milestone/multi-user:default' />
+  </dependency>
+
+  <exec_method type='method' name='start'
+    exec='ctrun -l child -o noorphan,regent /opt/oxide/installinator/bin/installinator install --from-ipcc --install-on-gimlet --stay-alive &amp;'
```

Comment:
For my own knowledge: what is the ampersand at the end for? Is that shell job control and if so what does it achieve in smf? Naively I'd expect that smf would want to wait for the process to exit rather than running it in the background.

Reply:
It is shell job control, yeah. There's a bit of cargo culting here, but experimentally: I believe SMF is expecting these methods (`start` here and `stop` below; others are also allowed like `refresh`) to exit indicating success/failure, and to possibly spawn a child process if necessary. If I remove the `&amp;` everything seems to still work, but I think only because `timeout_seconds='0'` means "infinite" - if I remove the `&amp;` and set `timeout_seconds` to a nonzero value, I see an error like this after that many seconds:

```
[ Mar  6 07:44:38 Method or service exit timed out.  Killing contract 224. ]
[ Mar  6 07:44:38 Method "start" failed due to signal KILL. ]
```

Reply:
That seems right.  With `timeout_seconds='0'`, things may appear to work at first but I expect SMF would believe the service to be stuck transitioning.  Instead of showing up in `svcs` as `online`, I'd expect it to say `offline*` (which means "offline but in-transition").